testmanage.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+import shutil
+import sys
+import warnings
+
+from django.core.management import execute_from_command_line
+
+
+os.environ["DJANGO_SETTINGS_MODULE"] = "wagtailsharing.tests.settings"
+
+
+def make_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--deprecation",
+        choices=["all", "pending", "imminent", "none"],
+        default="imminent",
+    )
+    return parser
+
+
+def parse_args(args=None):
+    return make_parser().parse_known_args(args)
+
+
+def runtests():
+    args, rest = parse_args()
+
+    only_wagtail = r"^wagtail(\.|$)"
+    if args.deprecation == "all":
+        # Show all deprecation warnings from all packages
+        warnings.simplefilter("default", DeprecationWarning)
+        warnings.simplefilter("default", PendingDeprecationWarning)
+    elif args.deprecation == "pending":
+        # Show all deprecation warnings from wagtail
+        warnings.filterwarnings(
+            "default", category=DeprecationWarning, module=only_wagtail
+        )
+        warnings.filterwarnings(
+            "default", category=PendingDeprecationWarning, module=only_wagtail
+        )
+    elif args.deprecation == "imminent":
+        # Show only imminent deprecation warnings from wagtail
+        warnings.filterwarnings(
+            "default", category=DeprecationWarning, module=only_wagtail
+        )
+    elif args.deprecation == "none":
+        # Deprecation warnings are ignored by default
+        pass
+
+    argv = [sys.argv[0]] + rest
+
+    try:
+        execute_from_command_line(argv)
+    finally:
+        from wagtail.tests.settings import STATIC_ROOT, MEDIA_ROOT
+
+        shutil.rmtree(STATIC_ROOT, ignore_errors=True)
+        shutil.rmtree(MEDIA_ROOT, ignore_errors=True)
+
+
+if __name__ == "__main__":
+    runtests()
```

Comment:
I don't understand the purpose of adding this file. Can you elaborate on its necessity?

If the intent is to add in deprecation warnings and fail on them, we already have a pattern we're using in another CFPB library, [Django-Flags](https://github.com/cfpb/django-flags/blob/main/tox.ini#L8) that I would prefer to see used, where we invoke `python` with:

```
python -W error::DeprecationWarning -W error::PendingDeprecationWarning
```

Reply:
Ah no reason other than I wanted to conveniently runs test locally without running tox. It's something I'm used to in our own packages. Happy to remove it 👍 and that was the plan.